### DISCUSSION TITLE
isLoading & isFetching

### DISCUSSION THREAD

**Discussion by qu-chen-x** [TanStack/query#4522] [Q&A]
### Describe the bug

As a junior front-end developer, I recently encountered a small problem when use react query:I realized that I don't really understand the essence of the isLoading & isFetching descriptions, so I hope I can get some help from you here. Do they mean the same thing? Also, is there an inclusive or sequential relationship between them?Can anyone give me an answer? Thank you very much.https://tanstack.com/query/v4/docs/reference/useQuery

### Your minimal, reproducible example

no

### Steps to reproduce

no

### Expected behavior

no

### Platform

no

### react-query version

no

### TypeScript version

no

**Comment by TkDodo** [comment]
have you read this page?
https://tanstack.com/query/v4/docs/guides/queries

**Comment by dante01yoon** [comment]
isFetching is about network connection - [Network Mode](https://tanstack.com/query/v4/docs/guides/network-mode#network-mode-online)

isLoading is about you have data of specific query key you passed in useQuery.
